terraform/modules/upload-service/dns.tf

```diff
@@ -0,0 +1,31 @@
+data "aws_route53_zone" "deployment_stage" {
+  name = "${var.parent_zone_domain_name}."
+}
+
+data "aws_acm_certificate" "deployment_stage" {
+  domain   = "*.${var.parent_zone_domain_name}"
+  statuses = ["ISSUED"]
+  types = ["AMAZON_ISSUED"]
+  most_recent = true
+}
+
+resource "aws_api_gateway_domain_name" "upload" {
```

Comment:
how is the custom domain linked to the api/lambda? Would this have to be manually done for new environments?

Reply:
So chalice currently handles the creation of your API Gateway (TM) rest API. I've stared at the chalice code for some time now and convinced myself that it builds a plan based on what resources currently exist in the AWS account, but cannot find the place where it hooks in the domain name or the call that wraps hooking that in (https://github.com/aws/chalice/blob/d4772db3decf40265ab089500d9ce8263fb8ac24/chalice/deploy/planner.py). I think the fastest path to confirming this is to run a deployment of the upload-service from scratch to see if chalice is smart enough to hook itself into the existing API Gateway domain name.

Thoughts?

Reply:
I think that makes sense. Approving this PR. I will actually be creating a new environment tomorrow to test something out so I can test this out while doing that